variables.tf

```
@@ -248,6 +253,29 @@ variable "encryption_at_host_enabled" {
   default = true
 }
 
+variable "entra_id_login" {
+  description = "Configuration for enabling EntraID-based login for virtual machines. Set 'enabled' to true to activate the feature and specify the list of 'principal_ids' that are allowed to log in. Note: This feature requires at least 1GB of memory, and certain SKUs like 'Standard_B1ls', 'Basic_A0', and 'Standard_A0' are not supported."
+  type = object({
+    enabled       = optional(bool)
+    principal_ids = optional(list(string))
+  })
+  default = ({
+    enabled       = false
+    principal_ids = []
+  })
```

`enabled` is required and defaults to false. `principal_ids` defaults to an empty list. This improves the documentation’s readability, in my opinion.

```suggestion
  type = object({
    enabled       = optional(bool)
    principal_ids = optional(list(string), [])
  })
  default = ({
    enabled       = false
  })
```